durabletask/worker.py

```diff
@@ -517,12 +564,32 @@ def process_event(self, ctx: _RuntimeOrchestrationContext, event: pb.HistoryEven
                             f"{ctx.instance_id}: Ignoring unexpected timerFired event with ID = {timer_id}.")
                     return
                 timer_task.complete(None)
-                ctx.resume()
+                if timer_task._retryable_parent is not None and timer_task._retryable_type is not None:
+                    activity_action = timer_task._retryable_parent._action
+                    
+                    if timer_task._retryable_type == "activity":
+                        cur_scheduled_task = activity_action.scheduleTask
+                        ctx.call_activity_util(id=activity_action.id, activity=cur_scheduled_task.name,
+                                                        input=cur_scheduled_task.input.value,
+                                                        retry_policy=timer_task._retryable_parent._retry_policy,
+                                                        act_task=timer_task._retryable_parent)
+                    elif timer_task._retryable_type == "sub_orch":
+                        cur_sub_orch = activity_action.createSubOrchestration
+                        ctx.call_sub_orchestrator_util(id=activity_action.id, orchestrator_name=cur_sub_orch.name,
+                                                        input=cur_sub_orch.input.value,
+                                                        instance_id=cur_sub_orch.instanceId,
+                                                        retry_policy=timer_task._retryable_parent._retry_policy,
+                                                        orch_task=timer_task._retryable_parent)
+                else:
+                    ctx.resume()
             elif event.HasField("taskScheduled"):
                 # This history event confirms that the activity execution was successfully scheduled.
                 # Remove the taskScheduled event from the pending action list so we don't schedule it again.
                 task_id = event.eventId
                 action = ctx._pending_actions.pop(task_id, None)
+                activity_task = ctx._pending_tasks.get(task_id, None)
+                if isinstance(activity_task, task.RetryableTask):
+                    activity_task.set_first_attempt(ctx.current_utc_datetime)
```

Comment:
Shouldn't we be setting this value inside the `call_activity` method, when we initially create the task?

Reply:
Yeah, that sounds better - have changed.